### ISSUE TITLE
Screen Reader Accessibility

### ISSUE THREAD

**Issue by Menelion** [miroslavpejic85/mirotalk#314]
### Describe the bug

Mirotalk seems to be a good calling solution but its interface is not accessible for blind and visually impaired users using screen reading software such as [JAWS for Windows](https://en.wikipedia.org/wiki/JAWS_(screen_reader) or [NVDA](https://en.wikipedia.org/wiki/NonVisual_Desktop_Access).  
After analyzing the source code, I can deduce that the main problem is that all the buttons lack proper labels.

### Steps to reproduce

1. Open [Mirotalk](https://p2p.mirotalk.com) with a screen reader running. For test the free and open-source [NVDA](https://www.nvaccess.org/download/) can be used.
2. Join a meeting by creating a new room.
3. Try to navigate the screen with arrow keys or Tab.

### Expected behavior

buttons like "Mute Microphone" or "Start Screen Share" must be announced by the screen reader.

### Device

- [ ] Android
- [ ] iPhone
- [ ] Linux
- [ ] Mac
- [x] Windows

### Browser

- [x] Chrome (or Chromium based)
- [ ] Firefox
- [ ] Safari
- [ ] Android browser
- [ ] iOS browser
- [ ] Other desktop browser

### Relevant log output

```shell
N/A
```

### Reproducibility

Yes

### Additional context

After examining the source code, I can see it's quite easy to grasp even for a mostly backend guy like me, so I'm willing to help in fixing this.

**Comment by miroslavpejic85** [comment]
Hey @Menelion ,

Rather than a bug, this looks more like a feature request 😉
By the way, I’ve started working on a module to handle all of this, here are the snippets.

[screenReader.js](https://github.com/user-attachments/files/23440523/screenReader.js)

**Comment by miroslavpejic85** [comment]
@Menelion I still need to test it on Windows with **NVDA**, but on macOS the built-in **VoiceOver** (Command + F5) correctly reads all button labels, since we’re using **Tippy** to set them. You can see a demo here: [VoiceOver Demo](https://nextcloud.cloudron.pocketsolution.net/s/3oe64aq8TRCcDYF).

Can you confirm whether this isn’t the issue here, or if **NVDA** might not be able to read them?

If **NVDA** can read them properly, then the only thing missing would be announcing certain events to the screen reader, for example, when the microphone or camera is turned on or off, and similar state changes eg:

> buttons like "Mute Microphone" or "Start Screen Share" must be announced by the screen reader.

![Image](https://github.com/user-attachments/assets/a414fb38-afcd-4b7e-8606-df986608cf77)

**Comment by miroslavpejic85** [comment]
PS: Windows has a **built-in screen reader called Narrator**, which is included with the operating system and requires no extra download or installation. It speaks aloud what is on your screen, and you can activate and deactivate it by pressing the **Windows logo key + Ctrl + Enter**. Did you tried it as well?

**Comment by miroslavpejic85** [comment]
That’s awesome! Your experience really shows and you’re absolutely right, JAWS and NVDA seems better.

I kept **`screenReader.js`** as simple as possible, it only handles announcing messages.

I also added the most important accessibility improvements you mentioned in this task, along with a few others I thought were useful.

Feel free to suggest more if you think something’s missing. The usage format is simple:

```js
screenReaderAccessibility.announceMessage('Message to announce');
```

By default, the message is announced with **polite** priority.
If you want it to be **assertive**, just pass it as a second parameter:

```js
screenReaderAccessibility.announceMessage('Message to announce', 'assertive');
```

**Priority parameter:**

* **Default:** `'polite'` non-interrupting, waits for the user to finish listening.
* **Optional:** `'assertive'`, interrupts the current speech for urgent messages.

Let me know if this feels better now or if you think it could use further improvements. Feel free to open a PR as well!

Thanks again for your time and feedback!
